From e9105fa205e7983b79e7d75ce95973705e34b3d9 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Fri, 20 Feb 2026 05:24:46 +0000
Subject: [PATCH 200/213] WIP: arm64: dts: rockchip: rk3576: Add IEP2 and IOMMU
 nodes

Add the IEP2 image enhancement processor at 0x27960000 and its
companion IOMMU at 0x27960800 for the RK3576 SoC. IEP2 provides
hardware-accelerated YUV deinterlacing. IEP2 uses VDPP clock
and reset resources and shares GIC_SPI 333 between the IEP2 and
its IOMMU.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3576.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3576.dtsi b/arch/arm64/boot/dts/rockchip/rk3576.dtsi
index f6291bc8f605..5c97e1ae6556 100644
--- a/arch/arm64/boot/dts/rockchip/rk3576.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3576.dtsi
@@ -1360,6 +1360,30 @@ gpu: gpu@27800000 {
 			status = "disabled";
 		};
 
+		iep2: iep@27960000 {
+			compatible = "rockchip,rk3576-iep2", "rockchip,rk3568-iep2";
+			reg = <0x0 0x27960000 0x0 0x500>;
+			interrupts = <GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cru ACLK_VDPP>, <&cru HCLK_VDPP>,
+				 <&cru CLK_CORE_VDPP>;
+			clock-names = "aclk", "hclk", "sclk";
+			iommus = <&iep2_mmu>;
+			power-domains = <&power RK3576_PD_VPU>;
+			resets = <&cru SRST_A_VDPP>, <&cru SRST_H_VDPP>,
+				 <&cru SRST_CORE_VDPP>;
+			reset-names = "axi", "ahb", "core";
+		};
+
+		iep2_mmu: iommu@27960800 {
+			compatible = "rockchip,rk3576-iommu", "rockchip,rk3568-iommu";
+			reg = <0x0 0x27960800 0x0 0x40>;
+			interrupts = <GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cru ACLK_VDPP>, <&cru HCLK_VDPP>;
+			clock-names = "aclk", "iface";
+			power-domains = <&power RK3576_PD_VPU>;
+			#iommu-cells = <0>;
+		};
+
 		vdec: video-codec@27b00000 {
 			compatible = "rockchip,rk3576-vdec";
 			reg = <0x0 0x27b00000 0x0 0x100>,
-- 
2.43.0

